### PR TITLE
Add autoplay and loop support to YouTube component

### PR DESCRIPTION
**Pull request by snehankekre**
## 📚 Context

The Docs team is trialing moving some GIFs over to YouTube. Those GIFs autoplayed in a loop. To preserve the same behavior when converted to YouTube videos, this PR lets authors optionally set the autoplay and looping behavior.

## 🧠 Description of Changes

- Refactor component to conditionally render captions
- Add autoplay and loop props to component
- Sets the video to muted when the video loops or autoplays (to avoid annoying users and avoid making them question if the video is supposed to have audio)

The changes of this PR can be pulled into #861 which includes a similar change.

## 💥 Impact

Size:

- [x] Small 
- [ ] Not small 

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

File: components/blocks/youTube.js [hunk 1]
```diff
@@ -2,41 +2,33 @@ import React from "react";
...
-    );
-  } else {
-    YouTubeBlock = (
+  if (autoplay || loop) {
+    mute = 1;
+  }
+
+  YouTubeBlock = (
+    <section className={styles.Container}>
       <section className={styles.IframeContainer}>
```
Comment (sfc-gh-jgarcia) [inline components/blocks/youTube.js:15]: Can we change this to a `<div>` component, for semantic's purpose?

File: components/blocks/youTube.js [hunk 2]
```diff
@@ -2,41 +2,33 @@ import React from "react";
...
           title="YouTube video player"
           frameBorder="0"
           allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
           allowFullScreen
         ></iframe>
       </section>
-    );
-  }
+      {caption && (
+        <section className={styles.CaptionContainer}>
```
Comment (sfc-gh-jgarcia) [inline components/blocks/youTube.js:26]: Similarly, can we do `<div>` here?

**Review by sfc-gh-jgarcia — APPROVED** [review]
LGTM besides two minor nits 👍 is there a video where we can QA the behavior?

File: components/blocks/youTube.js (hunk 1 above)
Comment (snehankekre) [reply components/blocks/youTube.js:15]: Good call! Fixed on 331de0 I hope

**Comment by snehankekre** [comment]
> is there a video where we can QA the behavior?

As a demo, I temporarily added a caption and enabled autoplay and loop to the video on `/library/advanced-features/connecting-to-data`.

**Comment by sfc-gh-jgarcia** [comment]
> > is there a video where we can QA the behavior?
> 
> As a demo, I temporarily added a caption and enabled autoplay and loop to the video on `/library/advanced-features/connecting-to-data`. Should be available at https://deploy-preview-862--streamlit-docs.netlify.app/library/advanced-features/connecting-to-data once the tests pass.

Looks and works great across all major desktop and mobile clients 👍

File: components/blocks/youTube.js (hunk 1 above)
Comment (sfc-gh-jgarcia) [reply components/blocks/youTube.js:15]: I was actually referring to the inner container, so more like this:

```
<section className={styles.Container}>
  <div className={styles.IframeContainer}>
    ...
  </div>
</section>
```

Sorry if it wasn't clear enough!

Comment (snehankekre) [reply components/blocks/youTube.js:15]: 😬 My bad! Creating a new PR now

**Pull request merged**